src/corehost/cli/args.cpp

```diff
@@ -54,6 +54,17 @@ void setup_shared_store_paths(const hostpolicy_init_t& init, const pal::string_t
         append_path(&args->dotnet_shared_store, RUNTIME_STORE_DIRECTORY_NAME);
         append_path(&args->dotnet_shared_store, init.tfm.c_str());
     }
+    
+    // "dotnet.exe" relative shared store folder, adding the arch so that unzipped developer scenario works
+    if (init.host_mode == host_mode_t::muxer)
```

Comment:
This is to support developers who download a zipped cli to try out this feature.
I have merge the condition, the reason for adding the path, is at cli runtime store creation time we always add architecture to the output path